### PR TITLE
Update changelog, and simplify PR and issue template

### PR DESCRIPTION
**Pull request by josh146**
**Description of the Change:**

* Updates the changelog

* Updates the PyPI project web address to point to the GitHub repository, so that GitHub dependency tracking works properly.

* ~Updates the CI to use the latest version of TF in the tests (1.13.1 -> 1.14.0)~ See https://github.com/XanaduAI/pennylane/pull/284#issuecomment-517597208.

* Simplifies the issue and PR template, making them clearer and more concise.

* Adds a line to the PR template, reminding users to add an entry to the changelog.

**Benefits:** all the above

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

**Comment by josh146** [comment]
Does anyone have any idea why `test_about()` causes a crash on Python 3.6+? I can recreate this locally as well, but I've never had this issue previously.

**Comment by josh146** [comment]
I can confirm the crash is related to `pip` installing TensorFlow 1.14. For some reason, TF 1.14 cases PyTest tests with `capfd` and `caplog` fixtures to crash with no traceback.

Reverted to running the test suite with TF==1.13.1.

**Comment by josh146** [comment]
I believe this issue is related to https://github.com/tensorflow/tensorflow/issues/29842.

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -1,60 +1,38 @@
...
+If posting a PennyLane issue, delete everything above the dashed line, and fill
+in the template.
 
-If making a feature request, delete the following template and describe, in detail, the feature and why it is needed.
+If making a feature request, delete the following template and describe, in detail,
+the feature and why it is needed.
 
-For general technical details:
-* Check out our documentation: https://pennylane.readthedocs.io
+For general technical details. check out our documentation:
```
Comment (johannesjmeyer) [inline .github/ISSUE_TEMPLATE.md]:
```suggestion
For general technical details check out our documentation:
```

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -1,60 +1,38 @@
...
-
-* **Installation method:**
-
-  Did you install PennyLane via pip, or directly from the GitHub repository source code?
-
-  If installed via GitHub, what branch/commit did you use? You can get this information by opening a terminal in the
-  PennyLane source code directory, and pasting the output of:
-  git log -n 1 --pretty=format:"%H%n%an%n%ad%n%s"
-
+* *System information:* (post the output of `import pennylane as qml; qml.about()`)
```
Comment (johannesjmeyer) [inline .github/ISSUE_TEMPLATE.md]: :+1:

**Pull request merged**